### PR TITLE
添加通过查询 UC 的 /v4/query API 实现动态加载服务器地址

### PR DESCRIPTION
**Pull request by bachue**

### CODE REVIEW

File: syncdata/operation/download.go [hunk 1]
```diff
@@ -6,27 +6,55 @@ import (
...
 
+var downloadClient = &http.Client{
+	Transport: &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   1 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
```
Comment (longbai) [inline syncdata/operation/download.go:26]: why use http2 in private network?

File: syncdata/operation/query.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+var queryClient = &http.Client{
+	Transport: &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   500 * time.Millisecond,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
```
Comment (longbai) [inline syncdata/operation/query.go:23]: http2?

File: syncdata/operation/query.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+		} else {
+			urls[i] = fmt.Sprintf("http://%s", domain)
+		}
+	}
+	return urls
+}
+
+func (queryer *Queryer) query() (*cache, error) {
+	var err error
+	c := queryer.getCache()
```
Comment (longbai) [inline syncdata/operation/query.go:122]: should lock get op

Comment (bachue) [reply syncdata/operation/query.go:122]: @longbai `cacheMap` is a thread-safe hashmap

**Review by longbai — APPROVED** [review]
LGTM

**Pull request merged**